GVFS/GVFS.Platform.Mac/MacDaemonManager.cs

```diff
@@ -0,0 +1,79 @@
+using GVFS.Common;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Platform.Mac
+{
+    /// <summary>
+    /// Class to query the configured services on macOS
+    /// </summary>
+    public class MacDaemonManager
+    {
+        private const string LaunchCtlPath = @"/bin/launchctl";
+        private const string LaunchCtlArg = @"list";
+
+        private IProcessHelper processHelper;
+
+        public MacDaemonManager()
+        : this(new ProcessHelperImpl())
+        {
+        }
+
+        public MacDaemonManager(IProcessHelper processHelper)
+        {
+            this.processHelper = processHelper;
+        }
+
+        public bool TryGetDaemons(string currentUser, out List<DaemonInfo> daemons, out string error)
+        {
+            // Consider for future improvement:
+            // Use Launchtl to run Launchctl as the "real" user, so we can get the process list from the user.
+            ProcessResult result = this.processHelper.Run(LaunchCtlPath, "asuser " + currentUser + " "  + LaunchCtlPath + " " + LaunchCtlArg, true);
+
+            if (result.ExitCode != 0)
+            {
+                error = result.Output;
+                daemons = null;
+                return false;
+            }
+
+            return this.TryParseOutput(result.Output, out daemons, out error);
+        }
+
+        private bool TryParseOutput(string output, out List<DaemonInfo> daemonInfos, out string error)
+        {
+            daemonInfos = new List<DaemonInfo>();
+
+            // 1st line is the header, skip it
+            foreach (string line in output.Split(new string[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries).Skip(1))
+            {
+                // The expected output is a list of tab delimited entried:
+                // PID\tSTATUS\tLABEL

```

Comment:
How likely is the output of this to change?

Reply:
I think this is stable. I thought about following up with a functional test...